processor/routingprocessor/factory_test.go

```diff
@@ -234,3 +234,20 @@ type mockProcessor struct{}
 func (mp *mockProcessor) processTraces(context.Context, ptrace.Traces) (ptrace.Traces, error) {
 	return ptrace.NewTraces(), nil
 }
+
+type mockHost struct {
```

Comment:
One thing I always wished was that our noop component test helpers would be more like mocks. Perhaps that would be a good contribution to the core repo?

https://github.com/open-telemetry/opentelemetry-collector/blob/e092fc728b724f862d37294dff03a602996f1dd2/component/componenttest/nop_host.go#L26-L28

Reply:
I can open a ticket for that.